### PR TITLE
Vtex/Product Details Page - Prevent malformed breadcrumbs when a category name contains slashes

### PR DESCRIPTION
**Pull request by AlanVncs**
When a categoy name contains slashes, the [current process](https://github.com/deco-cx/apps/blob/29d770f70f4107a8dd01bb6505273709dad5fcc6/vtex/utils/transform.ts#L393) will build a malformed breadcrumb.

image

image

The new aproach solves that problem.

### CODE REVIEW

**Review by guitavano — COMMENTED** [review]
Can you test without regex?

gpt suggestion:
```js
const names = categories.toReversed().reduce((acc, path, index) => {
  // Remove barras do início e do fim do caminho, se houver
  const trimmedPath = path.startsWith('/') ? path.slice(1) : path;
  const finalPath = trimmedPath.endsWith('/') ? trimmedPath.slice(0, -1) : trimmedPath;

  // Se for a primeira iteração, apenas adicione o caminho ao acumulador
  if (index === 0) return [finalPath];

  // Obtenha o último item do acumulador para comparação
  const lastItem = acc[index - 1];

  // Extraia a parte do caminho que vem após o último item
  const newItem = finalPath.includes(lastItem) ? finalPath.split(lastItem)[1] : finalPath;

  // Adicione o novo item ao acumulador
  return [...acc, newItem];
}, []);
```

**Comment by guitavano** [comment]
UP @AlanVncs

**Comment by guitavano** [comment]
https://www.atau.com.br/vitrine-seca-gourmet-120-polofrio/p?skuId=3832

File: vtex/utils/transform.ts [hunk 1]
```diff
@@ -463,7 +463,30 @@ const toBreadcrumbList = (
...
+    // Se for a primeira iteração, apenas adicione o caminho ao acumulador
+    if (index === 0) return [finalPath];
+
+    // Obtenha o último item do acumulador para comparação
+    const lastItem = acc[index - 1];
+
+    // Extraia a parte do caminho que vem após o último item
+    const newItem = finalPath.includes(lastItem)
+      ? finalPath.split(lastItem)[1]
+      : finalPath;
```
Comment (matheusgr) [inline vtex/utils/transform.ts:484]: @AlanVncs there is an issue with the last split:

image

**Review by matheusgr — DISMISSED** [review]

**Review by guitavano — APPROVED** [review]

**Pull request merged**